api/src/main/java/openconsensus/trace/SpanBuilder.java

```diff
@@ -110,6 +112,28 @@
  */
 public abstract class SpanBuilder {
 
+  /**
+   * Sets an attribute to the {@code Span}. If the {@code Span} previously contained a mapping for
+   * the key, the old value is replaced by the specified value.
+   *
+   * @param key the key for this attribute.
+   * @param value the value for this attribute.
+   * @return this.
+   * @since 0.1.0
+   */
+  public abstract SpanBuilder setAttribute(String key, AttributeValue value);
```

Comment:
Why do we need this? it seems that we can do this as the first thing after we create the Span. And in case of OT shim we can keep the "tags" around until we call startSpan on the main API then call addAttribute.

Reply:
My only question would be - why not? It will make easier to end user to migrate if there is this API on span builder